src/Agent.Worker/JobRunner.cs

```diff
@@ -279,6 +279,27 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     }
                 }
 
+#if OS_WINDOWS
+            bool retainDefaultEncoding = jobContext.Variables.GetBoolean(Constants.Variables.Agent.RetainDefaultEncoding) ?? false;
+            if (!retainDefaultEncoding) {
+                using (var p = HostContext.CreateService<IProcessInvoker>()) {
+                    int exitCode = p.ExecuteAsync(workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Root),
+                                             fileName: WhichUtil.Which("chcp", true, Trace),
+                                             arguments: "65001",
+                                             environment: null,
+                                             cancellationToken: CancellationToken.None).GetAwaiter().GetResult();
+                    if (exitCode == 0) {
+                        Trace.Info("Successfully changed to code page 65001 (UTF8)");
+                        Console.OutputEncoding = Encoding.UTF8;
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"'chcp 65001' failed with exit code {exitCode}");
```

Comment:
do we want to fail the build only because we can't print out UTF-8 char?
or should we just log a warning/info and continune?